pkg/builder/builder.go

```diff
@@ -181,9 +176,6 @@ func (b generalBuilder) Interpret() error {
 }
 
 func (b generalBuilder) compile(ctx context.Context) (*llb.Definition, error) {
-	if err := b.Interpret(); err != nil {
```

Comment:
Where do we run the interpreter if we removed this?

Reply:
It was moved to https://github.com/aseaday/envd/blob/75080a42d483900473f5e217f78743c26270c2d3/pkg/app/build.go#L127.
I do this because the def interpret need to be used to provide information about how to run the container. But in some cases like cache, the builder just need to interpret the manifest without needs to build the image.

Reply:
Gotcha, while `envd run` also requires an interpretation, thus maybe we should add the logic to the run command too.

Reply:
https://github.com/tensorchord/envd/blob/main/pkg/app/run.go#L103

Should we also need to refactor this section?

Reply:
Yes, but as for this, I am thinking about DefaultGraph or manifest's def should be accessible as a IR or  a wrap of interpreter itself. And the IR pass what builder need to builder. I am writing some docs to talk about it. I can refactor this after we all agree the design.
BTW, in this PR, I don't want a huge commit because kemingy also need a fix base on this PR.

Reply:
SGTM